Sources/HAP/Base/Accessory.swift

```diff
@@ -30,6 +35,29 @@ open class Accessory {
     public let type: AccessoryType
     public let info: Service.Info
     internal let services: [Service]
+    
+    // An accessory implementation can set this flag to false if a device becomes unreachable for a period
+    // If the accessory has a BridgingState Service, then the reachable property of that Service is set
+    //
+    open var reachable = true {
+        didSet {
+            if self.reachable != oldValue,
+               let bridgingStateService = services.first(where: { $0.type == .bridgingState }) as! Service.BridgingState? {
```

Comment:
The `as!` might fail, as having a `type == .bridgingState` does not require an instance of `BridgingState`. So best to make the code all generic, and query the generic accessory for the expected characteristic.

Reply:
Very good point. I'll address this. BTW, why does the setValue() func in Characteristic protocol have a fromConnection parameter ? It doesn't seem to be used.

Reply:
The parameter is used in `characteristics().swift` and makes sure that the originating connection doesn't get updated of a change it made itself.